### ISSUE TITLE
The exception in code causes ASP.NET thread to terminate.

### ISSUE THREAD

**Issue by wxtsxt** [severisv/SlackLogger#6]
To replicate the problem do this and log anything from a controller action
```
   logging.AddSlack(options =>
   {
       options.WebhookUrl = webhookUrl;
       options.SanitizeOutputFunction = output =>
       {
           throw new Exception("logslack-error");
       };
   });
```
The process terminates. The problem is this code:
`private async void PostAsync`
It is an async method NOT returning a Task.

To explain it better see the code versions below with a private asynch method 

Version I
I have and async method that throws exception
```
        private async Task ThrowExceptionAsync()
        {
            throw new Exception("");
        }
```
I call it using async/await pattern (from an asynch controller action)
```
            try
            {
                await ThrowExceptionAsync();
            }
            catch (Exception e)
            {
                Console.WriteLine("caught");
            }
            Console.WriteLine("out of try-catch");
```
The output is
_caught
out of try-catch_
 
 
Version II
I call my async method from a controller synchronous method:
```
            try
            {
                ThrowExceptionAsync();
            }
            catch (Exception e)
            {
                Console.WriteLine("caught");
            }
            Console.WriteLine("out of try-catch");
```

The output is
_out of try-catch_

Version III
Now the async method returns void instead of Task. I get the same output, but the asp.net thread process fails 
_at Controller.ThrowExceptionAsync()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
C:\Program Files\dotnet\dotnet.exe (process 18736) exited with code 0._

The implications of using this approach is explained here https://stackoverflow.com/a/15523793/5393059
I see this code fixed in https://github.com/mcristales/Gestionix.SlackLogger/blob/master/Util.SlackLogger/SlackService.cs, but it causes the logger to wait until the post to Slack ends, which is not exactly what we want.

The easiest fix is to make the method `PostAsync` not async
I could make a fix if you wish. Do you have any tests for the library?

**Comment by severisv** [comment]
Even if we made it return a task it would still be fire-and-forget here, wouldn't it?

https://github.com/severisv/SlackLogger/blob/90f9d2510e23be4ec591ab0510b607073b0ec9b0/SlackLogger/Logger.cs#L43

Wouldn't we still have the same problem then?

Could an alternative be to try/catch here? 
https://github.com/severisv/SlackLogger/blob/90f9d2510e23be4ec591ab0510b607073b0ec9b0/SlackLogger/SlackService.cs#L170


Either way, if you are able to make a fix for it that would be fantastic 🙂

Unfortunately, I haven't made any tests for this lib - been testing manually here: https://github.com/severisv/SlackLogger/tree/master/SlackLogger.Web.2

**Comment by wxtsxt** [comment]
When I return a task, the code is still fire-and-forget (no await), but the process does not break.
It is possible to do a try-catch to cover the whole body of the async method, but then what to do with the caught exception? I would leave it for now.
I need this change to start using this package and I have not found a better one, so I will gladly help out. Watch out for at least two merge requests within next two weeks.

**Comment by phillijw** [comment]
I think this may be the cause of a problem I'm having as well. My code calls `LogCritical("...")` and it reaches the `PostAsync` code but never actually manages to send anything (as far as I can tell). If I put a delay after the log call, it will successfully send to slack. I have a feeling this is related to not using Task in this code.

I'm going to try and see if I can make some improvements with this.

**Comment by phillijw** [comment]
I *temporarily* pushed out a package you can test with if you want. I will be deprecating this once @severisv is able to push out a new version though.

JoePhilly.SlackLogger Version=2.0.10

**Comment by severisv** [comment]
This is fixed in `2.0.11`. Thanks to @phillijw!
